File: common/src/main/java/me/ycdev/android/demo/ble/common/DeviceInfo.java
package me.ycdev.android.demo.ble.common;

import android.bluetooth.le.ScanResult;

import java.util.Comparator;

public class DeviceInfo {
    public int number;
    public int foundCount;
    public String btAddress;
    public ScanResult scanResult;

    public DeviceInfo(int number, String btAddress, ScanResult scanResult) {
        this.number = number;
        this.foundCount = 1;
        this.btAddress = btAddress;
        this.scanResult = scanResult;
    }

    public static class TimestampComparator implements Comparator<DeviceInfo> {
        @Override
        public int compare(DeviceInfo lhs, DeviceInfo rhs) {
            if (lhs.scanResult.getTimestampNanos() < rhs.scanResult.getTimestampNanos()) {
                return 1;
            } else if (lhs.scanResult.getTimestampNanos() > rhs.scanResult.getTimestampNanos()) {
                return -1;
            }
            return 0;
        }

        @Override
        public boolean equals(Object obj) {
            return super.equals(obj);
        }
    }
}

File: common/src/main/java/me/ycdev/android/demo/ble/common/DeviceInfo.kt
package me.ycdev.android.demo.ble.common

import android.bluetooth.le.ScanResult

import java.util.Comparator

class DeviceInfo(var number: Int, var btAddress: String, var scanResult: ScanResult) {
    var foundCount: Int = 0

    init {
        this.foundCount = 1
    }

    class TimestampComparator : Comparator<DeviceInfo> {
        override fun compare(lhs: DeviceInfo, rhs: DeviceInfo): Int {
            if (lhs.scanResult.timestampNanos < rhs.scanResult.timestampNanos) {
                return 1
            } else if (lhs.scanResult.timestampNanos > rhs.scanResult.timestampNanos) {
                return -1
            }
            return 0
        }
    }
}
